### ISSUE TITLE
@see tag only to allow a anchor element and not others

### ISSUE THREAD

**Issue by ps-sp** [checkstyle/checkstyle#5036]
cat test.javadoc
```
* @see <span>link</span>
```

java -jar master-8.2-all.jar -j test.javadoc
``` 
JAVADOC -> JAVADOC [0:0]
|--LEADING_ASTERISK -> * [0:0]
|--WS ->   [0:1]
|--JAVADOC_TAG -> JAVADOC_TAG [0:2]
|   |--SEE_LITERAL -> @see [0:2]
|   |--WS ->   [0:6]
|   |--HTML_ELEMENT -> HTML_ELEMENT [0:7]
|   |   `--HTML_TAG -> HTML_TAG [0:7]
|   |       |--HTML_ELEMENT_START -> HTML_ELEMENT_START [0:7]
|   |       |   |--START -> < [0:7]
|   |       |   |--HTML_TAG_NAME -> span [0:8]
|   |       |   `--END -> > [0:12]
|   |       |--TEXT -> link [0:13]
|   |       `--HTML_ELEMENT_END -> HTML_ELEMENT_END [0:17]
|   |           |--START -> < [0:17]
|   |           |--SLASH -> / [0:18]
|   |           |--HTML_TAG_NAME -> span [0:19]
|   |           `--END -> > [0:23]
|   `--NEWLINE -> \r\n
`--EOF -> <EOF> [1:0]
```

Such trees should not be parsed. Perhaps a parse error can be preferred over parsing such tree.

**Comment by Krish-876** [comment]
Hello, I would like to work on this issue.
I will first reproduce the current behavior and inspect the Javadoc grammar/tests for `@see` handling.

**Comment by romani** [comment]
@Krish-876, a lot has changed from time issue opened.
Can you share CLI output of full valid  javadoc block that is part of compiled java file.
And share CLI output of javadoc tool and screenshot of generated html page for this tag

**Comment by Krish-876** [comment]
**`SeeTagDemo.java`:**
```java
package com.example;
/**
 * Demo class for see anchor tag.
 *
 * @see <a href="https://checkstyle.org">Checkstyle</a>
 */
public class SeeTagDemo {

    /**
     * A sample method.
     *
     * @see <a href="https://example.com">External Resource</a>
     */
    public void sampleMethod() {}
}
```

**Released version (checkstyle-10.21.0) `-T` output:**
```
COMPILATION_UNIT -> COMPILATION_UNIT [1:0]
|--PACKAGE_DEF -> package [1:0]
|   |--ANNOTATIONS -> ANNOTATIONS [1:11]
|   |--DOT -> . [1:11]
|   |   |--IDENT -> com [1:8]
|   |   `--IDENT -> example [1:12]
|   `--SEMI -> ; [1:19]
`--CLASS_DEF -> CLASS_DEF [8:0]
    |--MODIFIERS -> MODIFIERS [8:0]
    |   |--BLOCK_COMMENT_BEGIN -> /* [3:0]
    |   |   |--COMMENT_CONTENT -> *\n * Demo class for see anchor tag.\n *\n * @see <a href="https://checkstyle.org">Checkstyle</a>\n  [3:2]
    |   |   `--BLOCK_COMMENT_END -> */ [7:1]
    |   `--LITERAL_PUBLIC -> public [8:0]
    |--LITERAL_CLASS -> class [8:7]
    |--IDENT -> SeeTagDemo [8:13]
    `--OBJBLOCK -> OBJBLOCK [8:24]
        |--LCURLY -> { [8:24]
        |--METHOD_DEF -> METHOD_DEF [15:4]
        |   |--MODIFIERS -> MODIFIERS [15:4]
        |   |   |--BLOCK_COMMENT_BEGIN -> /* [10:4]
        |   |   |   |--COMMENT_CONTENT -> *\n     * A sample method.\n     *\n     * @see <a href="https://example.com">External Resource</a>\n      [10:6]
        |   |   |   `--BLOCK_COMMENT_END -> */ [14:5]
        |   |   `--LITERAL_PUBLIC -> public [15:4]
        |   |--TYPE -> TYPE [15:11]
        |   |   `--LITERAL_VOID -> void [15:11]
        |   |--IDENT -> sampleMethod [15:16]
        |   |--LPAREN -> ( [15:28]
        |   |--PARAMETERS -> PARAMETERS [15:29]
        |   |--RPAREN -> ) [15:29]
        |   `--SLIST -> { [15:31]
        |       `--RCURLY -> } [15:32]
        `--RCURLY -> } [16:0]
```

**Patched version (13.4.1-SNAPSHOT) `-T` output:**
```
COMPILATION_UNIT -> COMPILATION_UNIT [1:1]
|--PACKAGE_DEF -> package [1:1]
|   |--ANNOTATIONS -> ANNOTATIONS [1:12]
|   |--DOT -> . [1:12]
|   |   |--IDENT -> com [1:9]
|   |   `--IDENT -> example [1:13]
|   `--SEMI -> ; [1:20]
`--CLASS_DEF -> CLASS_DEF [8:1]
    |--MODIFIERS -> MODIFIERS [8:1]
    |   |--BLOCK_COMMENT_BEGIN -> /* [3:1]
    |   |   |--COMMENT_CONTENT -> *\n * Demo class for see anchor tag.\n *\n * @see <a href="https://checkstyle.org">Checkstyle</a>\n  [3:3]
    |   |   `--BLOCK_COMMENT_END -> */ [7:2]
    |   `--LITERAL_PUBLIC -> public [8:1]
    |--LITERAL_CLASS -> class [8:8]
    |--IDENT -> SeeTagDemo [8:14]
    `--OBJBLOCK -> OBJBLOCK [8:25]
        |--LCURLY -> { [8:25]
        |--METHOD_DEF -> METHOD_DEF [15:5]
        |   |--MODIFIERS -> MODIFIERS [15:5]
        |   |   |--BLOCK_COMMENT_BEGIN -> /* [10:5]
        |   |   |   |--COMMENT_CONTENT -> *\n     * A sample method.\n     *\n     * @see <a href="https://example.com">External Resource</a>\n      [10:7]
        |   |   |   `--BLOCK_COMMENT_END -> */ [14:6]
        |   |   `--LITERAL_PUBLIC -> public [15:5]
        |   |--TYPE -> TYPE [15:12]
        |   |   `--LITERAL_VOID -> void [15:12]
        |   |--IDENT -> sampleMethod [15:17]
        |   |--LPAREN -> ( [15:28]
        |   |--PARAMETERS -> PARAMETERS [15:29]
        |   |--RPAREN -> ) [15:29]
        |   `--SLIST -> { [15:31]
        |       `--RCURLY -> } [15:32]
        `--RCURLY -> } [16:1]
```

**Generated HTML** (screenshot attached):

<img width="2560" height="2312" alt="Image" src="https://github.com/user-attachments/assets/d5e6686f-4b56-46b6-8a18-ae182f67a25f" />


The `-T` output is identical between released and patched versions at the Java AST level — the Javadoc comment content is preserved correctly in both. The HTML renders both `@see` anchor tags as clickable hyperlinks.

**Comment by mahfouz72** [comment]
@Krish-876, I checked the CLI output and the Javadoc tool behavior for the example in the issue description (You checked a different example btw).

``` 
D:\CS\test cat src/Test.java
/**
 * @see <span>link</span>
 */
public class Test {
}
D:\CS\test javadoc src/Test.java
Loading source file src\Test.java...
Constructing Javadoc information...
Building index for all the packages and classes...
Standard Doclet version 21.0.2+13-LTS-58
Building tree for all the packages and classes...
src\Test.java:2: warning: no main description
 * @see <span>link</span>
   ^
Generating .\Test.html...
src\Test.java:4: warning: use of default constructor, which does not provide a comment
public class Test {
       ^
Generating .\package-summary.html...
Generating .\package-tree.html...
Generating .\overview-tree.html...
Building index for all classes...
Generating .\allclasses-index.html...
Generating .\allpackages-index.html...
Generating .\index-all.html...
Generating .\search.html...
Generating .\index.html...
Generating .\help-doc.html...
2 warnings
```

---

<img width="642" height="466" alt="Image" src="https://github.com/user-attachments/assets/b5164504-65c9-43be-bbb5-df3f865e65fa" />

---

The Javadoc tool doesn’t produce any error here. It parses the input and just renders it as plain text in the generated docs. So even though this doesn’t really make sense semantically, it’s still accepted.

Because of that, I don’t think we should be stricter than the Javadoc tool. Users might already have similar cases in their code, and rejecting them in our parser could break things for no real benefit.

So I think it’s better to stay aligned with Javadoc’s behavior and close this issue without any changes.

**Comment by Krish-876** [comment]
Thank you for checking this. I agree that the javadoc tool is lenient here. However, the Javadoc spec explicitly restricts `@see` to a string literal, a reference, or an `<a href>` tag .  Checkstyle often enforces spec correctness independently of tool leniency, would that reasoning still apply here?

**Comment by romani** [comment]
I agree with @mahfouz72
